### ISSUE TITLE
Desktop Wallet unable to retrieve/refresh genesys wallet info/balance on remote server node

### ISSUE THREAD

**Issue by vRobM** [ArkEcosystem/ark-deployer#13]
It seems that unless a peer is added with the IP address of the server and port, the wallet cannot function properly. No balance, no transactions possible.

This is a remote server install with wallet not on the same machine.

**Comment by alexbarnsley** [comment]
Are you saying desktop wallet doesn't work at all for sidechains? Is this only the case for remote sidechains?

**Comment by vRobM** [comment]
It appears the wallet needs a functional peer. If it's not able to connect to 127.0.0.[1-5]:port it won't be functional.
Hence adding a real IP peer solved the issue for me with the node running on a remote VPS.

If the sidechain was meant to run local only, document it. 

If not, please fix the peer issue.

**Comment by alexbarnsley** [comment]
That's fine, all I asked was a question for confirmation. It's not designed to just run locally. There's no need for the rude tone you gave in your response.

**Comment by alexbarnsley** [comment]
Hi @vRobM, I was able to get the desktop wallet to work without any problems. Here's some commands you can run to get this set up, but you can follow the screenshots below and give it a try yourself (the port is different to the screenshots - make sure you use 4100).

Commands you can run:
```
forever stopall #make sure no running nodes (there's a fix for this coming)
./sidechain.sh install-node --name ArkTest --database ark_test --token ARKTEST --symbol AT --prefix T --forgers 5 --blocktime 16 --transactions-per-block 500 --reward-height-start 0 --skip-deps
./sidechain.sh start-node --name ArkTest
```

Passphrase:
>alpha artwork grit flower invest arrange sentence girl slab change valid mosquito

API:
>http://104.225.219.35:4100

![image](https://user-images.githubusercontent.com/8069294/36218292-c53574da-11ab-11e8-86d4-96ccf4314424.png)
![image](https://user-images.githubusercontent.com/8069294/36218306-ce2f586c-11ab-11e8-9bcc-192f11605f37.png)
![image](https://user-images.githubusercontent.com/8069294/36218359-fd9a27c6-11ab-11e8-9ecb-c0f4a6ae13f7.png)
![image](https://user-images.githubusercontent.com/8069294/36218395-1daff068-11ac-11e8-8cff-61cf44fb5a74.png)
![image](https://user-images.githubusercontent.com/8069294/36218413-2d36a84c-11ac-11e8-81ff-dfeadec68671.png)
![image](https://user-images.githubusercontent.com/8069294/36218418-317ab20e-11ac-11e8-824f-e12514a7555f.png)
![image](https://user-images.githubusercontent.com/8069294/36218429-38365a94-11ac-11e8-9759-6bccbd958a6c.png)

I have re-built the chain since the above, but here's a final shot of network and wallet:

![image](https://user-images.githubusercontent.com/8069294/36218751-4e0ba74c-11ad-11e8-94c4-ad9b79911423.png)
![image](https://user-images.githubusercontent.com/8069294/36218759-536e98b6-11ad-11e8-8c6f-ebf0f90257ac.png)

It's worth noting the below section of the readme. If any of these are set, the wallet doesn't interract well with the network.

![image](https://user-images.githubusercontent.com/8069294/36218772-67eae27c-11ad-11e8-9686-c7ad09be386f.png)

**I plan on writing a guide for a simple setup but it's low on my priority list**

**Comment by vRobM** [comment]
Hi @alexbarnsley 
A guide would be great, showing how to start/stop the sidechains for folks. I had to look at the code.

Thanks for trying to validate, but in all your screen shots, I don't see the peer list. I see that the peer icon isn't red, and I don't see if the peer it's on is listed as 127.0.0.[1-5].  (it likely isn't.)

I don't need help getting the sidechain to run. I have it running already.  I also have a need to NOT run it on port 4100, hence the other PR for enabling setting a PORT during install.

The issue is with the default install and seemingly the local peer list.
If nothing is changed and the node runs in default config, it will only have the local peer addresses in the peer list. 
This doesn't seem to let the wallet connect to a peer even though it already has the seed server info which it could use as a peer itself.

Maybe the simplest ask is to have the seed server be a peer?

**Comment by vRobM** [comment]
Here's an example of a new deployment
![image](https://user-images.githubusercontent.com/2704733/36245238-08d41556-11df-11e8-9a51-05aee3ed9cde.png)

Here's a failure to refresh on the genesys address:
![image](https://user-images.githubusercontent.com/2704733/36245315-6af5e016-11df-11e8-9bed-0d3ebe3d66c7.png)

ports are correct, chain:4020 explorer:4021

**Comment by alexbarnsley** [comment]
Not at my machine yet so not had chance to look just yet. Can you add a screenshot of your network too please?

**Comment by vRobM** [comment]
As requested:
![image](https://user-images.githubusercontent.com/2704733/36245815-c84df1e8-11e1-11e8-9757-e01abb73be2f.png)

**Comment by alexbarnsley** [comment]
Thanks for that. The only difference I made was I ticked "force" which forces the wallet to use the seed as a peer. This is necessary until actual forging delegates are added to the chain, as it's currently in an autoforging state where the only delegates are connected using loopback addresses. There's nothing wrong with the Deployer, Sidechain or the Wallet. It's just how the Ark network works, the sidechain is started in the autoforging state, forging nodes then need adding to the network in order for the peers list to grow. You fix the scenario you are talking about by adding new nodes to the network, using the config.CCC.json (basically how ARKCommander starts a node), just the same way as the Ark network runs.
